### DISCUSSION TITLE
CIM ingest with European

### DISCUSSION THREAD

**Discussion by Rayentorres** [GRIDAPPSD/GOSS-GridAPPS-D#1790] [General]
I am trying to use a CIM file for ingesting a new feeder into GridApps-D. I am using CIM 100 but with multiple files and following an european format (see the example).  Which are the main requirements for the CIM file for being compatible with GridApps? Do you have some tool for converting between different CIM formats? 
[DIGIN10-30-MV.zip](https://github.com/user-attachments/files/17920280/DIGIN10-30-MV.zip)

**Comment by shpoudel** [comment]
@Rayentorres - Thanks for reaching out. 

For the GridAPPS-D project, we use OpenDSS's cim100 command to generate CIM models. However, the platform supports CIM files created using OpenDSS version 1.2.17 and other compatible versions, but not the latest ones. It appears that the CIM models referenced in the message were generated using a more recent version of OpenDSS.

This specific verison of OpenDSS (linux version, i.e., opendsscmd) can be found here: https://github.com/GRIDAPPSD/GOSS-GridAPPS-D/tree/master/opendss

**Reply by rayento** [reply]
Thanks for your answer, Shiva. I have installed the old version of OpenDSS, but I haven't been able to generate the CIM file. Do you have an example in this version of OpenDSS where I can generate the CIM file?

**Comment by AAndersn** [comment]
@Rayentorres -- We have not tested the GridAPPS-D platform with CGMES CIM models, such as DIGIN. 

In theory, you should be able to load each of the model .xml files using 

```bash
curl -s -D- -H 'Content-Type: application/xml' --upload-file './path/model_piece.xml' -X POST 'http://localhost:8889/bigdata/namespace/kb/sparql'
```
You can then access the models through the PowerGrid Models API or CIMantic Graphs (https://github.com/PNNL-CIM-Tools/CIM-Graph/tree/develop). I am planning on adding JSON-LD support based on the DIGIN10 and other files, but that is probably a couple months out.

However, we can't guarantee that the simulations will run with these models as the European models may have attributes not supported by the power flow solver conversion.

**Reply by rayento** [reply]
Thanks for your answer Alex.  I will load DIGIN using the command line you suggested. I let you know if the simulation is able to run or not.

**Comment by theeglaze** [comment]
Hey Internet, 

I found the GridAppsd platform from a research paper I read on DERMS.

Currently, I am trying to set up the application on my laptop with the instructions I found on: 

 https://gridappsd-training.readthedocs.io/en/latest/api_usage/3.3-Using-the-PowerGrid-Models-API.html#Available-Models-in-Default-Installation

It seems the content is kinda out of date, and as I am not a pro at coding, I resorted to using AI to assist in the setup process; however, this has been largely a fruitless endeavour. 

It seems that ./run.sh works as expected, but if followed by docker logs -f gridapps, the command line returns, "waiting on connection". Different AI platforms have recommended creative ways of bypassing this message and getting to ./run-gridappsd.sh; however, brute forcing doesn't seem like the best course of action. 

Any help on getting the platform up and running in conjunction with the gridappsd-distributed-sample-app would be greatly appreciated.

P.S. My ultimate research goal is to: 

Bring automated intelligence to the DG /DES location; hence, coordination of the resources can be accomplished onsite instead of via SCADA and System Control at a separate location. This ideally reduces the communication requirement as the onsite intelligence (really a distributed energy resource management system - DERMS) updates system control periodically on its current state and the predicted future state of the DG/DES.
